### PR TITLE
Supports multiple selection of line, rect, polygon tools

### PR DESCRIPTION
**Pull request by Glenfiddish**

### CODE REVIEW

File: packages/lb-annotation/src/core/toolOperation/Selection.ts [hunk 1]
```diff
@@ -0,0 +1,186 @@
...
+  public hardSetSelectedIDs(ids: string[]) {
+    this.selectedIDs = ids;
+  }
+
+  public selectAll() {
+    this.selectedIDs = this.visibleDataList.map((i) => i.id);
+    this.toolInstance.render();
+  }
+
+  public toStashDataList() {
```
Comment (Kerwin-L) [inline packages/lb-annotation/src/core/toolOperation/Selection.ts:121]: Please add more comment

File: packages/lb-annotation/src/core/toolOperation/LineToolOperation.ts [hunk 2]
```diff
@@ -1978,14 +1895,14 @@ class LineToolOperation extends BasicToolOperation {
   }
 
   public renderTextAttribute() {
-    if (!this.ctx || !this.activeLine || this.activeLine?.length < 2 || this.isCreate) {
+    if (!this.ctx || !this.selectedLine) {
```
Comment (Kerwin-L) [inline packages/lb-annotation/src/core/toolOperation/LineToolOperation.ts:1898]: It needs to keep `this.activeLine?.length < 2`

File: packages/lb-annotation/src/core/toolOperation/polygonOperation.ts [hunk 3]
```diff
@@ -1157,6 +1180,92 @@ class PolygonOperation extends BasicToolOperation {
...
+      return;
+    }
+
+    const { selectedPolygon } = this;
+
+    if (!selectedPolygon) {
+      return;
+    }
+
+    /**
```
Comment (Kerwin-L) [inline packages/lb-annotation/src/core/toolOperation/polygonOperation.ts:1233]: The following code is `getDragOffset` that it is just return the  `offset`. You need to read `PolygonOperation.dragPolygon` to rewrite it.

File: packages/lb-annotation/src/core/toolOperation/rectOperation.ts [hunk 4]
```diff
@@ -656,15 +656,15 @@ class RectOperation extends BasicToolOperation {
...
-        }
+        // if (
+        //   this.dragInfo.dragTarget !== EDragTarget.Plane &&
+        //   (selectedRect.x < basicX - 0.01 ||
+        //     selectedRect.y < basicY - 0.01 ||
+        //     selectedRect.width > basicX + basicWidth - selectedRect.x + 0.01 ||
+        //     selectedRect.height > basicY + basicHeight - selectedRect.y + 0.01)
+        // ) {
+        //   return;
+        // }
```
Comment (Kerwin-L) [inline packages/lb-annotation/src/core/toolOperation/rectOperation.ts:667]: How can you to ensure the rect is in `drawOutsideTarget `?

File: packages/lb-annotation/src/core/toolOperation/polygonOperation.ts (hunk 3 above)
Comment (Glenfiddish) [reply packages/lb-annotation/src/core/toolOperation/polygonOperation.ts:1233]: What's the meaning?

File: packages/lb-annotation/src/core/toolOperation/rectOperation.ts (hunk 4 above)
Comment (Glenfiddish) [reply packages/lb-annotation/src/core/toolOperation/rectOperation.ts:667]: image

**Pull request merged**